# creating our own custom exception class by inheriting from the exception class
# The sys library will have all the information about any exception getting controlled
# exc_tb will have all the information like on which file the exception has occurred, on which line etc.
# the information will be provided by exc_info function

import sys
from src.logger import logging


def error_message_detail(error,error_detail:sys):
    _,_,exc_tb = error_detail.exc_info()
    file_name = exc_tb.tb_frame.f_code.co_filename

    error_message = "Error occurred in python script name [{0}] line number [{1}] error message [{2}]".format(
        file_name, exc_tb.tb_lineno, str(error)
    )

    return error_message

class CustomException(Exception):
    
    def __init__(self, error_message, error_detail:sys):
        super().__init__(error_message)
        self.error_message = error_message_detail(error_message, error_detail=error_detail)

    

    def __str__(self):
        return self.error_message